File: Data Structure/9.Hash Table/First Unique Character in a String.java
// https://leetcode.com/problems/first-unique-character-in-a-string/
class Solution {
    public int firstUniqChar(String s) {
        
        int [] arr = new int[26];
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            int index = c - 97;
            arr[index]+=1;
        }
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i);
            
            int index = c - 97;
            if(arr[index]==1){
                return i;
            }
           
        }
        
        
        return -1;
    }
}

File: Data Structure/9.Hash Table/first_unique_character_in_a_string.py
# https://leetcode.com/problems/first-unique-character-in-a-string/
class Solution:
    def firstUniqChar(self, s: str) -> int:
        
        count = collections.Counter(s)
        
        for i in s:
            if count[i] == 1:
                return s.index(i)
        return -1
        
        
#         n = len(s)
#         arr = [0]*26
        
#         ans = 9999999
#         for i in s:
#             ind = ord(i)-97
#             arr[ind] += 1

            
#         ans = 99999999
#         for i in s:
#             ind = ord(i)-97
#             if arr[ind]==1:
#                 ans = min(ans,s.index(i))
        
#         if ans==9999999:
#             return -1
#         return ans
            